src/Controller/DownloadController.php

```diff
@@ -108,6 +108,9 @@ public function defaultAction(int $id)
     public function httpAction(Dataset $dataset, Datastore $dataStore, LogActionItemEventDispatcher $logActionItemEventDispatcher)
     {
         $datasetSubmission = $dataset->getDatasetSubmission();
+        if ($dataset->isRestricted()) {
+            throw new BadRequestHttpException('Unable to download restricted dataset');
```

Comment:
Now that I look at it, this should have probably been a forbidden type exception.